### ISSUE TITLE
Backtracking in REPL

### ISSUE THREAD

**Issue by dram** [trealla-lang/trealla#99]
Backtracking in REPL seems to be broken, query completed immediately without prompting for other values:

```
% ./tpl 
Trealla v0.1alpha, 64-bits (int=128-bits), Feb 15 2017
?- between(1, 5, N).
 N: 1 
(0.000) true.
?- 
```

**Comment by dram** [comment]
Not sure if it is related, another REPL problem mentioned in #79 still exists:

```
% tpl samples/queens8.pro 
Trealla v0.1alpha, 64-bits (int=128-bits), Feb 15 2017
INFO: consulted 'samples/queens8.pro': 240 terms
?- rangeList(1, 8, Ns).

(0.000) false.
?- 
```

**Comment by infradig** [comment]
Fixed:

$ ./tpl
Trealla v0.1alpha, Feb 27 2017
?- between(1, 5, N).
 N: 1 (abort=a,trace=t,next=;): 
 N: 2 (abort=a,trace=t,next=;): 
 N: 3 (abort=a,trace=t,next=;): 
 N: 4 (abort=a,trace=t,next=;): 
 N: 5 (abort=a,trace=t,next=;): 

(0.000) false.
?- halt.
Halted

The rangelist problem don't know yet.

**Comment by infradig** [comment]
Finally.

**Comment by infradig** [comment]
This fix breaks #76

**Comment by infradig** [comment]
problem was with query_dump of vars.
